trieTools/zeroBalanceSystemAccountChecker/main.go

```diff
@@ -0,0 +1,409 @@
+		if err != nil {
+			return nil, nil, err
+		}
+
+		addressTokensMapInCurrFile, err := getFileContent(filepath.Join(fullPath, file.Name()))
+		if err != nil {
+			return nil, nil, err
+		}
+
+		shardAddressTokensMap[shardID] = copyMap(addressTokensMapInCurrFile)
+		merge(globalAddressTokensMap, addressTokensMapInCurrFile)
+
+		log.Info("read data from",
+			"file", file.Name(),
+			"shard", shardID,
+			"num tokens in shard", trieToolsCommon.GetNumTokens(shardAddressTokensMap[shardID]),
+			"num addresses in shard", len(shardAddressTokensMap[shardID]),
+			"total num addresses in all shards", len(globalAddressTokensMap))
+	}
+
+	return globalAddressTokensMap, shardAddressTokensMap, nil
+}
+
+func getShardID(file string) (uint32, error) {
+	shardIDStr := strings.TrimPrefix(file, "shard")
+	shardIDStr = strings.TrimSuffix(shardIDStr, ".json")
+	shardID, err := strconv.Atoi(shardIDStr)
+	if err != nil {
+		return 0, fmt.Errorf("invalid file input name; expected tokens shard file name to be <shardX.json>, where X = number(e.g. shard0.json)")
+	}
+
+	return uint32(shardID), nil
+}
+
+func getFileContent(file string) (map[string]map[string]struct{}, error) {
+	jsonFile, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+
+	bytesFromJson, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return nil, err
+	}
+
+	addressTokensMapInCurrFile := make(map[string]map[string]struct{})
+	err = json.Unmarshal(bytesFromJson, &addressTokensMapInCurrFile)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make(map[string]map[string]struct{})
+	for address, tokens := range addressTokensMapInCurrFile {
+		ret[address] = make(map[string]struct{})
+		for token := range tokens {
+			addTokenInMapIfHasNonce(token, ret[address])
+		}
+	}
+
+	return ret, nil
+}
+
+func addTokenInMapIfHasNonce(token string, tokens map[string]struct{}) {
+	if hasNonce(token) {
+		tokens[token] = struct{}{}
+	}
+}
+
+func hasNonce(token string) bool {
+	return strings.Count(token, "-") == 2
+}
+
+func copyMap(addressTokensMap map[string]map[string]struct{}) map[string]map[string]struct{} {
+	addressTokensMapCopy := make(map[string]map[string]struct{})
+
+	for address, tokens := range addressTokensMap {
+		addressTokensMapCopy[address] = make(map[string]struct{})
+		for token := range tokens {
+			addressTokensMapCopy[address][token] = struct{}{}
+		}
+	}
+
+	return addressTokensMapCopy
+}
+
+func merge(dest, src map[string]map[string]struct{}) {
+	for addressSrc, tokensSrc := range src {
+		_, existsInDest := dest[addressSrc]
+		if !existsInDest {
+			dest[addressSrc] = tokensSrc
+		} else {
+			log.Debug("same address found in multiple files", "address", addressSrc)
+			addTokensInDestAddress(tokensSrc, dest, addressSrc)
+		}
+	}
+}
+
+func addTokensInDestAddress(tokens map[string]struct{}, dest map[string]map[string]struct{}, address string) {
+	for token := range tokens {
+		dest[address][token] = struct{}{}
+	}
+}
+
+func exportSystemAccZeroTokensBalances(
```

Comment:
create a proper struct with this export functionality? Why do we need these nested maps?

Reply:
Created a struct with this func